### PR TITLE
Some naming convention and decoupling updates.

### PR DESCRIPTION
**Pull request by stratease**
[TEC-4516](https://theeventscalendar.atlassian.net/browse/TEC-4516)

Working on some more centralizing, and naming convention updates. In order to be nimble only doing some decoupling of the functionality - a good opportunity in a v2.

### CODE REVIEW

**Comment by andrasguseo** [comment]
@stratease This looks like at least a level above my level 🙂 so I will let Stephen have the honors.
One reco, the readme.md file can be deleted, it's usually not part of the extension.

File: src/TEC/Plugin.php [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+	 * Stores the base slug for the plugin.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @var string
+	 */
+	const SLUG = TEC_EXTENSION_WPAI_ADD_ON_SLUG;
+
+	/**
+	 * Stores the base slug for the extension.
```
Comment (Camwyn) [inline src/TEC/Plugin.php]:
```suggestion
	 * Stores the file for the extension.
```

File: src/TEC/Plugin.php [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+	const VERSION = '1.0.0';
+
+	/**
+	 * Stores the base slug for the plugin.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @var string
+	 */
+	const SLUG = TEC_EXTENSION_WPAI_ADD_ON_SLUG;
```
Comment (Camwyn) [inline src/TEC/Plugin.php]: Do we use this anywhere?

Comment (Camwyn) [reply src/TEC/Plugin.php]: Also, what's the benefit of storing a constant in a constant? Shorter string?

File: src/TEC/Plugin.php [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+		$this->container->register( PUE::class );
+
+		if ( ! $this->check_plugin_dependencies() ) {
+			// If the plugin dependency manifest is not met, then bail and stop here.
+			return;
+		}
+
+		// Do the settings.
+		// TODO: Remove if not using settings
+		//$this->get_settings();
```
Comment (Camwyn) [inline src/TEC/Plugin.php]: Ticket this or actually remove it?

File: src/TEC/Plugin.php [hunk 4]
```diff
@@ -0,0 +1,392 @@
...
+	private function get_options_prefix(): string {
+		return (string) str_replace( '-', '_', 'tec-labs-wpai-add-on' );
+	}
+
+	/**
+	 * Get Settings instance.
+	 *
+	 * @return Settings
+	 *
+	 * TODO: Remove if not using settings
```
Comment (Camwyn) [inline src/TEC/Plugin.php]: Looks like we're using settings here (and below), should we remove the TODOs?

**Review by Camwyn — CHANGES_REQUESTED** [review]
Mostly small stuff and a couple questions.

File: src/TEC/Plugin.php (hunk 2 above)
Comment (stratease) [reply src/TEC/Plugin.php]: Oh... no a side-effect of a refactor without thinking about it. I'll fix it. Good point.

File: src/TEC/Plugin.php (hunk 4 above)
Comment (andrasguseo) [reply src/TEC/Plugin.php]: While we could use some settings, i.e. retain blank meta entries, we don't necessarily have to. These are more for developers / professionals who can use the filters.

**Review by Camwyn — APPROVED** [review]

**Pull request merged**